File: weknot_android/app/src/main/java/com/example/weknot_android/network/comm/OpenChatComm.java
package com.example.weknot_android.network.comm;

import com.example.weknot_android.base.BaseComm;
import com.example.weknot_android.network.api.OpenChatApi;
import com.example.weknot_android.network.request.OpenChatRequest;
import com.example.weknot_android.room.entity.OpenChat.OpenChatRoom;
import com.example.weknot_android.room.entity.user.User;

import java.util.List;

import io.reactivex.Single;
import retrofit2.Response;

public class OpenChatComm extends BaseComm<OpenChatApi> {

    public Single<List<OpenChatRoom>> getChattingRooms(String token) {
        return api.getChattingRooms(token).map(getResponseObjectsFunction());
    }

    public Single<String> createChattingRoom(String token, OpenChatRequest openChatRequest) {
        return api.createChattingRoom(token, openChatRequest).map(Response::message);
    }

    public Single<List<User>> getChattingRoomUsers(String token, String roomNumber) {
        return api.getChattingRoomUsers(token, roomNumber).map(getResponseObjectsFunction());
    }

    @Override
    protected Class<OpenChatApi> type() {
        return OpenChatApi.class;
    }
}

File: weknot_android/app/src/main/java/com/example/weknot_android/network/comm/OpenChatComm.kt
package com.example.weknot_android.network.comm

import com.example.weknot_android.base.BaseComm
import com.example.weknot_android.model.chat.OpenChatRoom
import com.example.weknot_android.model.user.User
import com.example.weknot_android.network.api.OpenChatApi
import com.example.weknot_android.network.request.OpenChatRequest
import io.reactivex.Single

class OpenChatComm : BaseComm<OpenChatApi>() {

    fun getChattingRooms(token: String): Single<List<OpenChatRoom>> {
        return api.getChattingRooms(token).map(getResponseObjectsFunction())
    }

    fun createChattingRoom(token: String, openChatRequest: OpenChatRequest): Single<String> {
        return api.createChattingRoom(token, openChatRequest).map(getResponseMessageFunction())
    }

    fun getChattingRoomUsers(token: String, roomNumber: String): Single<List<User>> {
        return api.getChattingRoomUsers(token, roomNumber).map(getResponseObjectsFunction())
    }

    override fun type(): Class<OpenChatApi> {
        return OpenChatApi::class.java
    }
}